packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -803,18 +778,33 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
     _cachedAttributedValue = null;
     _cachedCombinedSemanticsInfos = null;
     _canComputeIntrinsicsCached = null;
-    markNeedsTextLayout();
+    markNeedsLayout();
     markNeedsSemanticsUpdate();
   }
 
+  TextPainter? _textIntrinsicsCache;
+  TextPainter get _textIntrinsics {
+    return (_textIntrinsicsCache ??= TextPainter())
+      ..text = _textPainter.text
+      ..textAlign = _textPainter.textAlign
+      ..textDirection = _textPainter.textDirection
+      ..textScaler = _textPainter.textScaler
+      ..maxLines = _textPainter.maxLines
+      ..ellipsis = _textPainter.ellipsis
+      ..locale = _textPainter.locale
+      ..strutStyle = _textPainter.strutStyle
+      ..textWidthBasis = _textPainter.textWidthBasis
+      ..textHeightBehavior = _textPainter.textHeightBehavior;
+  }
```

Comment:
I wonder if we could somehow have a test to verify that the two text painters remain in sync? I don't have any ideas of how to do that, though.

Reply:
I was hoping the dry size check in `RenderBox.size` setter would help catch inconsistencies in the two text painters?